packages/terra-data-grid/CHANGELOG.md

```diff
@@ -5,6 +5,7 @@
 * Fixed
   * Fixed issue where focus was given to the column header instead of its button element.
   * Fixed issue where row selection was being announced twice in Worklist Data Grid.
+  * Fixed an issue when data accesses invalid indexes when deleting rows and columns.
```

Comment:
The issue is not specific to rows or columns being "deleted".  The data for the table is being updated and the row and/or column index no longer exists.